### PR TITLE
Place the audio embeddings with one scatter

### PR DESCRIPTION
**Pull request by happyarts**
Fixes #4.

`_merge_input_embeddings` walked the prompt and tested `j in audio_positions` against a Python list for every token, then `mx.stack`ed `seq_length` one-row arrays. A Voxtral prompt is almost entirely audio — 375 placeholder tokens per 30 s — so that is quadratic in the thing that grows, on a path that runs for every `generate()`.

This does it with one scatter per batch row, which is what `transformers` does (`inputs_embeds.masked_scatter(...)`).

### Verification

Bit-identical output — every element, at each length below. M1 Max, `Voxtral-Mini-3B-2507` 8-bit, merge time with the encoder forward subtracted:

| audio | prompt tokens | before | after |
|---:|---:|---:|---:|
| 120 s | 1505 | 0.32 s | 0.09 s |
| 400 s | 5255 | 1.15 s | 0.10 s |
| 600 s | 7505 | 1.75 s | 0.09 s |

A real transcription through `apply_transcrition_request` + `generate()` was run to confirm the path is intact.

### The dtype promotion is deliberate

`embed_tokens` returns bf16, the projector returns float32, and the old code's closing `mx.stack` promoted the whole sequence to float32 as a side effect — which is why the audio embeddings kept their precision.

Scattering into the bf16 array instead rounds every audio embedding to bf16. My first attempt did that and came back 0.0078 off, which is bf16 epsilon at these magnitudes and would surface only as slightly different logits. Hence the explicit `astype` before the scatter; it is load-bearing, not tidying.

`numpy` is added as an import to `modeling_voxtral.py`. It is already a hard dependency (`audio_processing` imports it).

### CODE REVIEW

**Comment by happyarts** [comment]
Pushed one more commit after a review pass: the scatter writes in place, and unlike the previous `mx.stack` + `mx.concatenate` implementation that is visible through the caller's own reference.

```python
a = model.embed_tokens(ids).astype(mx.float32)
model._merge_input_embeddings(input_ids=ids, input_features=feats, inputs_embeds=a)
# `a` now contains the audio embeddings
```

The dtype promotion already copies whenever `embed_tokens`' bf16 meets the projector's float32 — which is every transcription — so this only bites a caller who hands in an array that is already float32. Guarded now, and the output stays bit-identical on the paths measured above.
